File: tlb/05/solutions/tac.pl
#!/usr/bin/perl -w

if (@ARGV == 0) { # no args, read from stdin
    print reverse <STDIN>;
}
else {
    foreach $file (@ARGV) {
        open my $input, '<', $file or die "$file:can not open: $!\n";
        print reverse <$input>;
        close $input;
    }
}

File: tlb/05/solutions/tac.py
#!/usr/bin/python

import sys

if len(sys.argv) == 1:
    for line in reversed(list(sys.stdin)):
        sys.stdout.write(line)
else:
    for filename in sys.argv[1:]:
        with open(filename) as f:
            for line in reversed(list(f)):
                sys.stdout.write(line)
